### DISCUSSION TITLE
onclick events data in salesforce

### DISCUSSION THREAD

**Discussion by rishachilles** [ghiscoding/slickgrid-universal#1032] [Q&A] [answered]
How do I fetch the data of the row which is clicked in salesforce? while accessing event?.detail?.args the args object only contains the row index and the column index data 

referring to examples tried event?.detail?.args?.dataContext but it shows as undefined.

Following is the output of printing  event?.detail?.args for onclick event :

{
  "row": 0,
  "cell": 6,
  "grid": {
    "slickGridVersion": "4.0.0",
    "onScroll": {},
    "onBeforeSort": {},
    "onSort": {},
    "onHeaderMouseEnter": {},
    "onHeaderMouseLeave": {},
    "onHeaderRowMouseEnter": {},
    "onHeaderRowMouseLeave": {},
    "onHeaderContextMenu": {},
    "onHeaderClick": {},
    "onHeaderCellRendered": {},
    "onBeforeHeaderCellDestroy": {},
    "onHeaderRowCellRendered": {},
    "onFooterRowCellRendered": {},
    "onFooterContextMenu": {},
    "onFooterClick": {},
    "onBeforeHeaderRowCellDestroy": {},
    "onBeforeFooterRowCellDestroy": {},
    "onMouseEnter": {},
    "onMouseLeave": {},
    "onClick": {},
    "onDblClick": {},
    "onContextMenu": {},
    "onKeyDown": {},
    "onAddNewRow": {},
    "onBeforeAppendCell": {},
    "onValidationError": {},
    "onViewportChanged": {},
    "onColumnsReordered": {},
    "onColumnsDrag": {},
    "onColumnsResized": {},
    "onColumnsResizeDblClick": {},
    "onBeforeColumnsResize": {},
    "onCellChange": {},
    "onCompositeEditorChange": {},
    "onBeforeEditCell": {},
    "onBeforeCellEditorDestroy": {},
    "onBeforeDestroy": {},
    "onActiveCellChanged": {},
    "onActiveCellPositionChanged": {},
    "onDragInit": {},
    "onDragStart": {},
    "onDrag": {},
    "onDragEnd": {},
    "onSelectedRowsChanged": {},
    "onCellCssStylesChanged": {},
    "onAutosizeColumns": {},
    "onBeforeSetColumns": {},
    "onBeforeUpdateColumns": {},
    "onRendered": {},
    "onSetOptions": {}
  }
}

**Comment by ghiscoding** [comment, marked as answer]
you can find the answer if you read the Wikis and search on Stack Overflow, some event only return the row/cell, then you need to use the DataView to get associated datacontext

see [Slickgrid, how to get the grid item on click event?](https://stackoverflow.com/questions/8365139/slickgrid-how-to-get-the-grid-item-on-click-event)

```ts
handleOnCellClicked(event) {
    const args = event?.detail?.args;
    const eventData = event?.detail?.eventData;
    const dataContext = this.sgb.dataView.getItem(args.row);
}
```

Please search through the Wikis and/or ask questions on Stack Overflow which is a better platform to ask questions. SlickGrid is used by thousands of people, there are other people who can answer your questions
